nixos/modules/services/x11/redshift.nix

```diff
@@ -19,18 +19,31 @@ in {
     };
 
     latitude = mkOption {
-      type = types.str;
+      type = types.nullOr types.str;
+      default = null;
       description = ''
         Your current latitude, between
-        <literal>-90.0</literal> and <literal>90.0</literal>.
+        <literal>-90.0</literal> and <literal>90.0</literal>. Must be provided
+        along with longitude.
       '';
     };
 
     longitude = mkOption {
-      type = types.str;
+      type = types.nullOr types.str;
+      default = null;
       description = ''
         Your current longitude, between
-        between <literal>-180.0</literal> and <literal>180.0</literal>.
+        between <literal>-180.0</literal> and <literal>180.0</literal>. Must be
+        provided along with latitude.
+      '';
+    };
+
+    provider = mkOption {
+      type = types.str;
+      default = "manual";
+      description = ''
+        The location provider to use for determining your location. Set to
+        <literal>manual</literal> if you are providing latitude/longitude.
```

Comment:
Also mention `geoclue2` as supported.

I suggest it's `enum` type of what we support since each provider needs extra Nix configuration.

Reply:
Yeah, the enum is better. Only the `manual` provider requires extra configuration, so I've kept that in the doc.